src/automations/index.ts

```diff
@@ -28,17 +39,27 @@ const invokeAdHocAutomation = (options: ICourierClientConfiguration) => {
 
 const invokeAutomationTemplate = (options: ICourierClientConfiguration) => {
   return async (
-    params: ICourierAutomationInvokeTemplateParams
+    params: ICourierAutomationInvokeTemplateParams,
+    config?: ICourierAutomationConfig
   ): Promise<ICourierAutomationInvokeResponse> => {
+    const axiosConfig: AxiosRequestConfig = {
+      headers: {},
+    };
+
+    if (config && config.idempotencyKey) {
+      axiosConfig.headers["Idempotency-Key"] = config.idempotencyKey;
+    }
```

Comment:
Do we need to something similar for `idempotencyExpiry`?

Reply:
Good cal. I missed that one